nixos/modules/system/boot/systemd-nspawn.nix

```diff
@@ -115,8 +134,16 @@ in {
       units = mapAttrs' (n: v: let nspawnFile = "${n}.nspawn"; in nameValuePair nspawnFile (instanceToUnit nspawnFile v)) cfg;
     in 
       mkMerge [
-        (mkIf (cfg != {}) { 
-          environment.etc."systemd/nspawn".source = mkIf (cfg != {}) (generateUnits "nspawn" units [] []);
+        (mkIf (cfg != {}) {
+          assertions = mapAttrsToList (n: v: {
+            assertion = v.restartOnChange -> !v.reloadOnChange;
+            message = ''
+              If ${n}.nspawn is supposed to be restarted on config change, it can't
+              be reloaded at the same time!
+            '';
+          }) cfg;
+          systemd.targets.machines.wants = map (x: "systemd-nspawn@${x}.service") (attrNames cfg);
```

Comment:
Not sure if we should do this by default.

Reply:
The upstream `systemd-nspawn@` unit has a `WantedBy=machines.target` so I think this is ok to do.

Reply:
I think it's a good idea.